cmd/restic/cmd_check.go

```diff
@@ -45,7 +46,11 @@ Exit status is 12 if the password is incorrect.
 	RunE: func(cmd *cobra.Command, args []string) error {
 		term, cancel := setupTermstatus()
 		defer cancel()
-		return runCheck(cmd.Context(), checkOptions, globalOptions, args, term)
+		summary, err := runCheck(cmd.Context(), checkOptions, globalOptions, args, term)
+		if globalOptions.JSON {
+			term.Print(ui.ToJSONString(summary))
```

Comment:
If check fails, then the summary should IMHO always return a non-zero number of errors. I'm pretty sure that everyone is going to check the exit code even though that's absolutely necessary.
```suggestion
			if err != nil && summary.NumErrors == 0 {
				summary.NumErrors = 1
			}
			term.Print(ui.ToJSONString(summary))
```

Reply:
Done.

_FYI: I had this in a previous version but then decided against it and only count errors arising from a broken repository but not from a failing check action._

Reply:
I'm aware it isn't ideal. But making sure that users noticed failed check runs has priority